BootloaderCommonPkg/Library/IppCryptoPerfLib/IppCryptoPerfLib.c

```diff
@@ -0,0 +1,137 @@
+
+/* @file
+*
+*  Copyright (c) 2024, Intel Corporation. All rights reserved.<BR>
+*
+*  SPDX-License-Identifier: BSD-2-Clause-Patent
+*
+*/
+
+#include <Library/CryptoLib.h>
+#include <Library/ShellLib.h>
+#include <Library/TimeStampLib.h>
+#include <Library/MemoryAllocationLib.h>
+#include <Library/SecureBootLib.h>
+
+
+#define KB_1   0x0400
+#define KB_8   0x2000
+#define KB_32  0x8000
+#define KB_64  0x10000
+#define KB_128 0x20000
+#define KB_256 0x40000
+#define MB_1   0x00100000
+#define MB_2   0x00200000
+#define MB_4   0x00400000
+#define MB_8   0x00800000
+#define MB_10  0x00A00000
+#define MB_16  0x01000000
+#define MAX_ITER 4
+#define MAX_DIGEST_BUFFER_SIZE SHA384_DIGEST_SIZE
+
+struct IterationLength {
+    UINT32 Length;
+    UINT8 Desc[7];
+};
+
+const struct IterationLength Iter_premem[] = {{ KB_1,   "1 KB"   },
+                                       { KB_32,  "32 KB"  },
+                                       { KB_64,  "64 KB"  },
+                                       { KB_128, "128 KB" }};
+
+const struct IterationLength Iter_postmem[] = {{ KB_64,  "64 KB"  },
+                                       { KB_256, "256 KB" },
+                                       { MB_1,   "1 MB"   },
+                                       { MB_10,  "10 MB"  }};
+
+enum Algorithm {
+    SHA256,
+    SHA384,
+    SM3,
+    MAX_ALGO
+};
+
+static const char* AlgorithmName (enum Algorithm Alg) {
+    if (Alg == SHA256)
+        return "SHA256";
+    else if (Alg == SHA384)
+        return "SHA384";
+    else if (Alg == SM3)
+        return "SM3";
+    else
+        return NULL;
+}
+
+VOID EFIAPI IppCryptoPerf_Premem()
+{
+    UINT64 StartTime, StopTime, TsTicks;
+    UINT8 DigestBuffer[MAX_DIGEST_BUFFER_SIZE];
+    UINT8* Data = (UINT8*)(UINTN)FixedPcdGet32(PcdStage1ABase);
+    UINT8 Index;
+    enum Algorith Alg;
+
+    ShellPrint (L"!!! IPP CRYPTO PERFORMANCE MESUREMENT - Pre-Mem!!!  \n");
+    ShellPrint (L" Usecase             | Delta (µs)      \n");
+    ShellPrint (L"---------------------+----------------+\n");
```

Comment:
Adding print would impact the total boot performance. Prefer using the standard way to record and show the performance data.

Reply:
done